pkg/cmd/server/origin/master.go

```diff
@@ -197,14 +198,22 @@ func (c *MasterConfig) InstallAPI(container *restful.Container) []string {
 	userEtcd := useretcd.New(c.EtcdHelper, user.NewDefaultUserInitStrategy())
 	oauthEtcd := oauthetcd.New(c.EtcdHelper)
 
+	// TODO: with sharding, this needs to be changed
 	deployConfigGenerator := &deployconfiggenerator.DeploymentConfigGenerator{
-		DeploymentInterface:       &clientDeploymentInterface{c.DeploymentClient()},
-		DeploymentConfigInterface: deployEtcd,
-		ImageRepositoryInterface:  imageEtcd,
+		Client: deployconfiggenerator.Client{
+			DCFn:   deployEtcd.GetDeploymentConfig,
+			IRFn:   imageEtcd.GetImageRepository,
+			LIRFn2: imageEtcd.ListImageRepositories,
+		},
```

Comment:
FYI, I also changed part of this here: https://github.com/openshift/origin/pull/648/files#diff-3999d64b0b8fa9423b47f9feb61db2b0L200